bundles/org.pitest.pitclipse.ui/src/org/pitest/pitclipse/ui/view/PitView.java

```diff
@@ -53,9 +98,57 @@ public void setFocus() {
     @Override
     public synchronized void update(File result) {
         if (result == null) {
+            homeUrlString = null;
             browser.setText("<html/>");
         } else {
-            browser.setUrl(result.toURI().toString());
+            homeUrlString = result.toURI().toString();
+            // fallback, if the file does not exist
```

Comment:
This comment refers to the result file or what? There doesn't seem to be any fallback here

Reply:
The fallback is in line 106. The normal index.html is opened and only a new url is set, if a new url points to an existing file.

Reply:
I'd avoid conditions that cannot be easily tested and in any case, setting the URL even if it's null or if the file does not exist should be safe.
Otherwise, if all the conditions can be covered by test then it's fine to me

Reply:
Should all be tested and I agree that it would be safe to set the url, but the user would get an "no such file" message in the browser.

Reply:
Well, if that's because there's no such file then the user should get that error, or I don't understand the scenario...